### ISSUE TITLE
HTTP ERROR 403 No valid crumb was included in the request when run by Harbormaster

### ISSUE THREAD

**Issue by zbrid** [google/llvm-premerge-checks#163]
**Describe the bug**
I'm trying to understand the Harbormaster build failure on my patch (https://reviews.llvm.org/D78390). 

Here is the build failure link: https://reviews.llvm.org/harbormaster/build/61415/1/?l=0

It appears the failure is unrelated to my patch since it looks like there were no builds or tests kicked off in the logs due to an HTTP request error. Am I misinterpreting this? Am I reading the wrong logs? 

This seems like it may be an error with the set up of this repo.

Relevant links:

**To Reproduce**
I don't know how to reproduce this as this was due to a run by Phabricator.

**Expected behavior**
I expected the error log for the failed build to show a failed test or a failed build command that could have been caused by my patch rather than an HTTP error.
**Screenshots**
NA

**Comment by zbrid** [comment]
Additional info before I restart the build: 

```

1 | HTTP 403
-- | --
Server: openresty/1.15.8.2
Date: Fri, 17 Apr 2020 18:53:57 GMT
Content-Type: text/html;charset=iso-8859-1
Content-Length: 593
Connection: keep-alive
Vary: Accept-Encoding
Strict-Transport-Security: max-age=15724800; includeSubDomains
X-Content-Type-Options: nosniff
Cache-Control: must-revalidate,no-cache,no-store

```

**Comment by zbrid** [comment]
Seen here on a different patch: https://reviews.llvm.org/harbormaster/build/61462/

**Comment by rudkx** [comment]
I'm seeing the same here: https://reviews.llvm.org/harbormaster/build/61575/

**Comment by ChristianKuehnel** [comment]
Thank you for finding and debugging this!

Jenkins seems to have installed some updates, increasing API security:
https://wiki.jenkins.io/display/JENKINS/CSRF+Protection

The solution is to use an API token instead of the password to trigger builds:
https://wiki.jenkins.io/display/JENKINS/Authenticating+scripted+clients

**Comment by ChristianKuehnel** [comment]
When debugging this I noticed a delay between hitting the button and a build actually being started:  #164

**Comment by ChristianKuehnel** [comment]
To trigger a build locally (with your username and password):
`curl -d '' "https://<user>:<password>@jenkins.llvm-merge-guard.org/job/amd64_debian_testing_clang8/buildWithParameters\?DIFF_ID\=258716\&PHID\=PHID-HMBT-wfa6cjatlpmona3gthky\&token\=kWJuaWIIZVqaX367vZIM61KLbyqOXBgIN2Xg9"`

Produces the exact same error message

**Comment by ChristianKuehnel** [comment]
The new version requires some more effort in triggering builds via the API call:
https://support.cloudbees.com/hc/en-us/articles/219257077-CSRF-Protection-Explained

However Harbormaster does not support that...

**Comment by ChristianKuehnel** [comment]
Found it:

We were referencing the container `jenkins/jenkins:lts` in our kubernetes [deployment](https://github.com/google/llvm-premerge-checks/blob/master/kubernetes/jenkins.yaml). For whatever reason, Kubernetes decided to restart (and pull) that container from the registry. This lead to an upgrade from version 2.190 to 2.222, as the `lts` tag was updated in the mean time in the repository.

In version 2.190 the CSRF was optional and could be disabled, in 2.222 it's always on and can't (easily) be disabled. In combination with the nginx reverse proxy used for authentication this leads to a situation where Phabricator can no longer user plain HTTP requests to trigger builds.

The short-term solution was to fix the Jenkins version to the tag `2.190.3` and to avoid automatic updates that way until we have found another solution (e.g. #161).

I just [re-enabled](https://reviews.llvm.org/H511) the builds for the BETA testers. If that keeps working, I'll enable the [builds](https://reviews.llvm.org/H576) for everyone tomorrow.
